src/hash.c

```diff
@@ -319,17 +318,10 @@ int verify_bundle_hash(struct manifest *manifest, struct file *bundle)
 				free(filename);
 				break;
 			}
-			free(filename);
 
-			string_or_die(&tar, TAR_COMMAND " -C %s/%i -xf %s/%i/Manifest.%s.tar 2> /dev/null",
-				      state_dir, current->last_change, state_dir,
-				      current->last_change, current->filename);
+			ret = extract_to(filename, state_dir);
```

Comment:
The manifest tars live in the versioned directories under statedir, not directly under statedir.